website/docs/language/functions/templatestring.mdx

```diff
@@ -0,0 +1,91 @@
+---
+sidebar_label: templatestring
+description: |-
+  The `templatestring` function takes a string and renders it as a template using a supplied set of template variables.
+---
+
+# `templatestring` Function
+
+The `templatestring` function takes a string and renders it as a template using a supplied set of template variables.
+
+```hcl
+templatestring(str, vars)
+```
```

Comment:
This is still missing a lot of documentation. This needs to be informative for the end users. Please see [templatefile.mdx](https://github.com/opentofu/opentofu/blob/199ec898e634b5c2782664cf436bac94eae3fd10/website/docs/language/functions/templatefile.mdx#L8-L43) as an example of this